src/server/models/tag.ts

```diff
@@ -0,0 +1,65 @@
+import Sequelize from 'sequelize'
+import { sequelize } from '../util/sequelize'
+import { IdType } from '../../types/server/models'
+import { TAG_KEY_REGEX, TAG_STRING_REGEX } from '../../shared/util/validation'
+
+export interface TagType extends IdType, Sequelize.Model {
+  readonly tagString: string
+  readonly tagKey: string
+}
+
+// For sequelize define
+type TagTypeStatic = typeof Sequelize.Model & {
+  new (values?: object, options?: Sequelize.BuildOptions): TagType
+}
+
+export const Tag = <TagTypeStatic>sequelize.define(
+  'tag',
+  {
+    tagString: {
+      type: Sequelize.STRING(25),
+      allowNull: false,
+      unique: true,
+      validate: {
+        is: TAG_STRING_REGEX,
+      },
+    },
+    tagKey: {
+      type: Sequelize.STRING(25),
+      allowNull: false,
+      validate: {
+        is: TAG_KEY_REGEX,
+      },
+    },
+  },
+  {
+    defaultScope: {
+      useMaster: true,
+    },
+    scopes: {
+      /**
+       * Use the replica database for read queries. To be enabled
+       * when realtime data is not needed.
+       */
+      useReplica: {
+        useMaster: undefined,
+      },
+    },
+    hooks: {
+      afterCreate: async (_: TagType, options) => {
```

Comment:
Can we add an inline comment here to clarify that it is necessary to check for a wrapper transaction because there's a downstream operation (assigning the tag to the url)?

Reply:
A tag is only created to be associated to a link. If somehow the association operation failed, it makes sense to remove the tag.